### PR TITLE
Updated tasks file 

### PR DESCRIPTION
**Pull request by roshanborkar01**
Created tasks file and copy pest the code for processAlgorithum.

### CODE REVIEW

**Comment by roshanborkar01** [comment]
In the views.py, the  `process_algorithm_async.delay(url, geojson)
` method should call in existing functions or create another to call it.Because it required arguments to get call.

File: Server/sprawl_calculator/wup_calculator/tasks.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    get_output_path,
+    calculate_and_save_wdis,
+    calculate,
+)
+from osgeo import gdal
+
+app = Celery('tasks', broker='redis://localhost:8000')
+
+@app.task
+def process_algorithm_async(url:str, geojson:GeoJSONFeature):
```
Comment (hrishiballal) [inline Server/sprawl_calculator/wup_calculator/tasks.py]: In celery, only strings can be passed so in this case we will pass only the processing_id and the first thing to do is to look up that in the DB and create a `AlgorithmProcessingParameters` object and then call the `generate_sprawl_indices` method.

Comment (hrishiballal) [reply Server/sprawl_calculator/wup_calculator/tasks.py]: For more information [refer how we do it](https://github.com/geodesignhub/urban-sprawl-calculation-service/blob/master/Server/sprawl_calculator/wup_calculator/views.py#L39) synchronously.

File: Server/sprawl_calculator/wup_calculator/tasks.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    '''
+    The "processAlgorithm" method is a part of the QGIS Processing Framework 
+    and it is used to perform some action for processing a raster and a vector layer.
+    '''
+
+    def processAlgorithm(url:str, geojson:GeoJSONFeature) -> WUPCalculationResult:
```
Comment (hrishiballal) [inline Server/sprawl_calculator/wup_calculator/tasks.py]: See above, this needs to re removed.

**Comment by roshanborkar01** [comment]
As per the next step to create a algorithm paramters object and the call the generate_sprawl_indices method is already there in [views.py](https://github.com/geodesignhub/urban-sprawl-calculation-service/blob/5441a4be7cfa3ad9248c539176933bbcaa23d4a3/Server/sprawl_calculator/wup_calculator/views.py#L39)

File: Server/sprawl_calculator/sprawl_calculator/settings.py [hunk 3]
```diff
@@ -123,3 +123,12 @@
 # https://docs.djangoproject.com/en/4.1/ref/settings/#default-auto-field
 
 DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
+
+# Celery settings
+CELERY_BROKER_URL = 'redis://localhost:8000'
+CELERY_RESULT_BACKEND = 'redis://localhost:8000'
```
Comment (hrishiballal) [inline Server/sprawl_calculator/sprawl_calculator/settings.py:129]: Celery broker and URL is on the Redis port, not Django one

File: Server/sprawl_calculator/sprawl_calculator/settings.py [hunk 4]
```diff
@@ -123,3 +123,12 @@
 # https://docs.djangoproject.com/en/4.1/ref/settings/#default-auto-field
 
 DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
+
+# Celery settings
+CELERY_BROKER_URL = 'redis://localhost:8000'
```
Comment (hrishiballal) [inline Server/sprawl_calculator/sprawl_calculator/settings.py:128]: Same as below, please fix the port and host

File: Server/sprawl_calculator/wup_calculator/views.py [hunk 5]
```diff
@@ -40,6 +41,8 @@ def wup_index_generator(request):
     # TODO: Make it async
     sprawl_index_generator.generate_sprawl_indices(parameters= algorithm_parameters)
 
+    process_algorithm_async.delay(url, geojson)
```
Comment (hrishiballal) [inline Server/sprawl_calculator/wup_calculator/views.py:44]: Lets put in the paramter names e.g. `url = url`

**Review by hrishiballal — CHANGES_REQUESTED** [review]

**Comment by hrishiballal** [comment]
We can close off this PR.

**Pull request closed without merging**